modP2p/test/org/aion/p2p/HeaderTest.java

```diff
@@ -1,23 +1,55 @@
+/*
+ * Copyright (c) 2017-2018 Aion foundation.
+ *
+ *     This file is part of the aion network project.
+ *
+ *     The aion network project is free software: you can redistribute it
+ *     and/or modify it under the terms of the GNU General Public License
+ *     as published by the Free Software Foundation, either version 3 of
+ *     the License, or any later version.
+ *
+ *     The aion network project is distributed in the hope that it will
+ *     be useful, but WITHOUT ANY WARRANTY; without even the implied
+ *     warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *     See the GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with the aion network project source files.
+ *     If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Contributors:
+ *     Aion foundation.
+ */
+
 package org.aion.p2p;
 
-import static junit.framework.Assert.assertEquals;
+import static org.junit.Assert.assertEquals;
 
-import java.io.IOException;
 import java.util.concurrent.ThreadLocalRandom;
+import org.junit.Before;
 import org.junit.Test;
 
 public class HeaderTest {
 
-    private short version = (short)ThreadLocalRandom.current().nextInt();
-    private byte ctl = 0;
-    private byte action = 4;
-    private int length = 8;
-    Header hd = new Header(version, ctl, action, length);
-    private int route = (version << 16) | (ctl << 8) | action;
+    private short version;
+    private byte ctl;
+    private byte action;
+    private int length;
+    private Header hd;
+    private int route;
+
+    @Before
+    public void setup() {
+        version = (short) ThreadLocalRandom.current().nextInt();
```

Comment:
can you print out the value of 'version' at the start of each test method so if it fails in CI server we can reproduce?

Reply:
revised in 9e87524535f62ab2528026c5168b540bcbfc4e9a